project/JARUtils.scala

```diff
@@ -219,4 +219,73 @@ object JARUtils {
     )
     IO.delete(tempJarPath.toFile)
   }
+
+  /** Reads the `Bundle-NativeCode` entries from the JAR manifest.
+    * See <a href="https://docs.osgi.org/specification/osgi.core/8.0.0/framework.module.html#framework.module-loading.native.code.libraries">
+    *   OSGi Bundle-NativeCode specification
+    * </a>
+    *
+    * If there is no such manifest attribute, an empty list is returned.
+    */
+  def readNativeCodeEntriesFromManifest(
+    jarPath: Path
+  ): List[NativeCodeEntry] = {
+    Using(new JarFile(jarPath.toFile)) { jarFile =>
```

Comment:
it's a bit hard to read this method as it is neither idiomatic Scala nor Java.
I think it would be fairly easy to rewrite it to the former by replacing all those extra returns and nulls with `Option`.